evalml/tests/component_tests/test_featuretools.py

```diff
@@ -163,3 +165,86 @@ def test_dfs_sets_max_depth_1(mock_dfs, X_y_multi):
     feature.fit(X_pd, y)
     _, kwargs = mock_dfs.call_args
     assert kwargs["max_depth"] == 1
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+def test_dfs_with_serialized_features(mock_dfs, X_y_binary):
+    X, y = X_y_binary
+    X_pd = pd.DataFrame(X)
+    X_pd.columns = X_pd.columns.astype(str)
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X", dataframe=X_pd, index="index", make_index=True
+    )
+    feature_matrix, features = ft.dfs(
+        entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
+    )
+
+    dfs = DFSTransformer(features=features)
+    dfs.fit(X_pd)  # no-op
+    assert not mock_dfs.called
+
+    X_t = dfs.transform(X_pd)
+    assert_frame_equal(feature_matrix, X_t)
+    assert features == dfs.features
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+@patch(
+    "evalml.pipelines.components.transformers.preprocessing.featuretools.calculate_feature_matrix"
+)
+def test_dfs_skip_transform(mock_calculate_feature_matrix, mock_dfs, X_y_binary):
+    X, y = X_y_binary
+    X_pd = pd.DataFrame(X)
+    X_pd.columns = X_pd.columns.astype(str)
+    X_fit = X_pd.iloc[: len(X) // 3]
+    X_transform = X_pd.iloc[len(X) // 3 :]
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X", dataframe=X_transform, index="index", make_index=True
+    )
+    feature_matrix, features = ft.dfs(
+        entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
+    )
+    features = list(filter(lambda f: not isinstance(f, IdentityFeature), features))
+    dfs = DFSTransformer(features=features)
+    dfs.fit(X_fit)  # no-op
+    X_t = dfs.transform(
+        feature_matrix
+    )  # no-op as well, feature_matrix contains features already
+    assert not mock_dfs.called
+    assert not mock_calculate_feature_matrix.called
+
+    assert_frame_equal(feature_matrix, X_t)
+    assert features == dfs.features
+
+
+@patch("evalml.pipelines.components.transformers.preprocessing.featuretools.dfs")
+def test_dfs_does_not_skip_transform_with_non_identity_feature(mock_dfs, X_y_binary):
```

Comment:
Added an additional test since previous tests only had identity features.

Reply:
I'm confused by this 😬  Both this test and `test_dfs_skip_transform` use :

```python
    feature_matrix, features = ft.dfs(
        entityset=es, target_dataframe_name="X", trans_primitives=["absolute"]
    )
    features = list(filter(lambda f: not isinstance(f, IdentityFeature), features))
```

Why does this test use `X_transform` to create the entity set as opposed to `X_pd` ?

Reply:
@jeremyliweishih and I spoke about this. My confusion was because I thought `non_identity_feature` referred to the `features` passed in init, but it refers to the features passed to `transform`. In this test, only the original features are passed to `transform` (X_pd) but in the previous case, the original + absolute features are passed to `transform`.